vite.config.mts

```diff
@@ -0,0 +1,152 @@
+
+const VITE_CSP_META = serializeCsp(cspMeta)
+
+const publicFilesEnvVars = {
+  VITE_CSP_META: JSON.stringify(VITE_CSP_META),
+}
+
+const publicPath = path.join(__dirname, 'public')
+
+for (const dirent of fs.readdirSync(publicPath, { withFileTypes: true })) {
+  if (!dirent.isFile()) continue
+  const mungedName = dirent.name
+    .toUpperCase()
+    .split('')
+    .map(x => (/^[0-9A-Z]$/.test(x) ? x : '_'))
+    .join('')
+  const data = fs.readFileSync(path.join(publicPath, dirent.name))
+
+  const integrity = ssri.fromData(data, {
+    strict: true,
+    algorithms: ['sha256'],
+  })
+  publicFilesEnvVars[`VITE_SRI_${mungedName}`] = JSON.stringify(integrity)
+
+  const digest = sha256.digest(data)
+  const cid = CID.create(1, raw.code, digest).toString()
+  publicFilesEnvVars[`VITE_CID_${mungedName}`] = JSON.stringify(cid)
+}
+
+// eslint-disable-next-line import/no-default-export
+export default defineConfig(({ mode }) => {
+  return {
+    plugins: [
+      react(),
+      tsconfigPaths(),
+      nodePolyfills({
+        globals: {
+          Buffer: true,
+          global: true,
+          process: true,
+        },
+        protocolImports: true,
+      }),
+      checker({
+        typescript: true,
+        overlay: true,
+      }),
+      process.env.ANALYZE === 'true' &&
+        analyzer({
+          analyzerMode: 'server',
+          analyzerPort: 8888,
+          openAnalyzer: true,
+        }),
+    ],
+    define: {
+      ...Object.fromEntries(
+        Object.entries(publicFilesEnvVars).map(([key, value]) => [`import.meta.env.${key}`, value]),
+      ),
+      ...Object.fromEntries(
+        Object.entries(publicFilesEnvVars).map(([key, value]) => [`process.env.${key}`, value]),
+      ),
+    },
+    server: {
+      port: 3000,
+      headers,
+    },
+    preview: {
+      port: 3000,
+      headers,
+    },
+    resolve: {
+      alias: {
+        '@': resolve(__dirname, './src'),
+        'ethers/lib/utils': 'ethers5/lib/utils.js',
+        'ethers/lib/utils.js': 'ethers5/lib/utils.js',
+        'dayjs/locale': resolve(__dirname, 'node_modules/dayjs/locale'),
+      },
+    },
+    build: {
+      target: 'esnext',
+      commonjsOptions: {
+        transformMixedEsModules: true,
+      },
+      chunkSizeWarningLimit: 2000,
+      rollupOptions: {
+        input: {
+          main: resolve(__dirname, 'index.html'),
+        },
+        output: {
+          manualChunks: id => {
+            if (id.includes('node_modules')) {
+              if (id.includes('react-icons')) return 'react-icons'
+              if (id.includes('dayjs')) return 'dayjs'
+              if (id.includes('lodash')) return 'lodash'
+              if (id.includes('@formatjs')) return '@formatjs'
+              if (id.includes('cosmjs-types')) return 'cosmjs-types'
+              return null
+            }
+            if (id.includes('src/assets/translations')) return 'translations'
+            return null
+          },
+        },
```

Comment:
Those will probably won't work at deploy time too as it will probably produce chunks with more than 26mb (didn't verify thought but I'm almost sure), will verify along with sourcemaps